lib/src/Uncertainty/Distribution/BernsteinCopulaFactory.cxx

```diff
@@ -105,31 +105,30 @@ Distribution BernsteinCopulaFactory::buildParallel(const Sample & empiricalCopul
   Mixture result(atomsMixture);
   // Here we know that the mixture is a copula even if none of its atoms is.
   result.isCopula_ = true;
-  return result;
+  return result.clone();
 }
 
-Distribution BernsteinCopulaFactory::buildSequential(const Sample & empiricalCopulaSample,
+BernsteinCopulaFactory::Implementation BernsteinCopulaFactory::buildSequential(const Sample & empiricalCopulaSample,
     const UnsignedInteger binNumber)
 {
   const UnsignedInteger size = empiricalCopulaSample.getSize();
   const UnsignedInteger dimension = empiricalCopulaSample.getDimension();
-  LOGINFO("BernsteinCopulaFactory - Create the resulting Bernstein copula");
   Mixture::DistributionCollection atomsMixture(size);
   for (UnsignedInteger i = 0; i < size; ++i)
   {
-    const Point nu(empiricalCopulaSample[i] * binNumber);
+    const Point nu(empiricalCopulaSample[i]);
     Mixture::DistributionCollection atomsKernel(dimension);
     for (UnsignedInteger j = 0; j < dimension; ++j)
-      atomsKernel[j] = Beta(std::floor(nu[j]) + 1.0, binNumber + 1.0, 0.0, 1.0);
+      atomsKernel[j] = Beta(std::floor(nu[j] * binNumber) + 1.0, binNumber + 1.0, 0.0, 1.0);
     atomsMixture[i] = ComposedDistribution(atomsKernel);
   }
   Mixture result(atomsMixture);
   // Here we know that the mixture is a copula even if none of its atoms is.
   result.isCopula_ = true;
-  return result;
+  return result.clone();
```

Comment:
We find this pattern everywhere in distribution factories, but I do not understand why we clone.  If we return a pointer, we should allocate it and return it, something like
```
  Pointer<Mixture> result(new Mixture(atomsMixture));
  // Here we know that the mixture is a copula even if none of its atoms is.
  result->isCopula_ = true;
  return result;
```

BTW this is a general remark, it can wait for OT 1.11.